app/composite-components/productsTabClient.tsx

```diff
@@ -0,0 +1,274 @@
+import {
+  Button,
+  BUTTON_TYPE, Details, FormCheckbox, FormSelect
+} from "@capgeminiuk/dcx-react-library";
+import { isEmpty } from "lodash";
+import { useEffect, useState } from 'react';
+import { AccessibleAutocomplete, ClientOnly } from '~/components';
+import { IErrorTransformed, ILabelAndValue, ISpecies } from "~/types";
+
+type ProductTabsProps = {
+  species: ISpecies[];
+  states: ILabelAndValue[];
+  presentations: ILabelAndValue[];
+  commodityCodes: ILabelAndValue[];
+  faoCode?: string;
+  stateCode?: string;
+  presentationCode?: string;
+  commodityCode?: string;
+  errors?: IErrorTransformed;
+};
+
+export const ProductsTabClient = ({
+  species,
+  states,
+  faoCode,
+  stateCode,
+  presentationCode,
+  presentations,
+  commodityCodes,
+  commodityCode,
+  errors
+}: ProductTabsProps) => {
+  const [selectedSpeciesCode, setSelectedSpeciesCode] = useState('');
+  const [selectedStateCode, setSelectedStateCode] = useState('');
+  const [selectedPresentationCode, setSelectedPresentationCode] = useState('');
+  const [speciesStateLookup, setSpeciesStateLookup] = useState([]);
+  const [speciesPresentationLookup, setSpeciesPresentationLookup] = useState([]);
+  const [speciesCommodityLookup, setSpeciesCommodityLookup] = useState([]);
+  const speciesName = (s)=> {
+    return `${s.faoName} (${s.faoCode})`;
+  }
+   useEffect(()=> {
+    if(selectedSpeciesCode) {
+      setSelectedStateCode('');
+     const fetchData = async ()=> {
+      const res = await (await fetch(`http://localhost:3001/reference/api/v1/speciesStateLookup?faoCode=${selectedSpeciesCode}`)).json();
```

Comment:
why double await?

Reply:
Double await for initial fetch then the .json() at the end. I can break into multiple lines for clarity.

Reply:
Still slightly confused why .json() need another one... is an async call?